### ISSUE TITLE
[BUG] issue in implementation of vPortYieldFromISR()  on AVR_AVRDx port

### ISSUE THREAD

**Issue by francesco-gritti** [FreeRTOS/FreeRTOS-Kernel-Partner-Supported-Ports#9]
as discussed on the FreeRTOS forum at https://forums.freertos.org/t/avr32-db-port-vportyieldfromisr-issue/17878
the current implementation of vPortYieldFromISR() is not usable on the AVR_AVRDx port.

considering the following interrupt handler>
```
ISR (i2c_VECT) {

    // do some checks and send data if needed

    if (all data have been sent) {
        xSemaphoreGiveFromISR(i2cSemaphore, NULL);
	    vPortYieldFromISR ();
    }
}
```

with the current implementation, which is

```
portSAVE_CONTEXT();
vTaskSwitchContext();
portRESTORE_CONTEXT();
asm volatile ( "reti" );
```

this would bring to a context save on the task stack of the interrupt context, which is undesired

The current fix for this issue is to declare interrupt handlers in the following way

```
__attribute__((naked)) ISR (i2c_VECT) {
    portSAVE_CONTEXT();

     // do some checks and send data if needed

    if (all data have been sent) {
        xSemaphoreGiveFromISR(i2cSemaphore, NULL);
	    vTaskSwitchContext();
	    portRESTORE_CONTEXT();
	    asm volatile ( "reti" );
    }

    portRESTORE_CONTEXT();
    asm volatile ("reti");
}
```

**Comment by aggarg** [comment]
@alexmchp Do you have any suggestion here? I think we should delete the following lines as these are not useful - 
* https://github.com/FreeRTOS/FreeRTOS-Kernel-Partner-Supported-Ports/blob/main/GCC/AVR_AVRDx/portmacro.h#L98-L99
* https://github.com/FreeRTOS/FreeRTOS-Kernel-Partner-Supported-Ports/blob/main/GCC/AVR_AVRDx/port.c#L160-L173

**Comment by aggarg** [comment]
@alexmhcp, do you plan to look at it?

**Comment by AniruddhaKanhere** [comment]
@aggarg can we make the above changes and fix the implementation? Or do we need a definitive answer from @alexchp?

**Comment by Skptak** [comment]
Marking this issue as closed as #13 removed the incorrect function.
Please feel free to raise a new issue or pull request to add the desired functionality to this port
Thanks!
